core/plugins/org.polarsys.capella.core.projection.exchanges/src/org/polarsys/capella/core/projection/exchanges/PhysicalLinksCreator.java

```diff
@@ -160,42 +165,70 @@ protected void createPhysicalLinksFromCExchanges(Component sourceContainer, Comp
         || doesNodeAlreadyHaveAPhysicalLinkForComponentExchange(sourceContainer, componentExchange)) {
       return;
     }
-    // get the opposite port [which could be source or target of the CE]
-    InformationsExchanger target = FunctionalExt.getOtherBound(componentExchange, port);
-    if (target instanceof ComponentPort) {
-      EObject targetContained = target.eContainer();
-      // find the target Node
-      if (targetContained instanceof Component && isValidBound((Component) targetContained)) {
-
-        if (targetContained instanceof PhysicalComponent) {
-          PhysicalComponent targetContainerPC = findDeployingComponent(sourceContainer, componentExchange,
-              (PhysicalComponent) targetContained);
-          if (targetContainerPC != null) {
-            doCreatePhysicalLink(componentExchange, sourceContainer, targetContainerPC);
-          }
+    Component targetContained = findTheTargetNode(componentExchange, port);
+    if (targetContained != null) {
+      if (targetContained instanceof PhysicalComponent) {
+        PhysicalComponent targetContainerPC = findDeployingComponent((PhysicalComponent) targetContained);
+        if (targetContainerPC != null) {
+          doCreatePhysicalLink(componentExchange, sourceContainer, targetContainerPC, port);
         }
-
-        // if the container is the same as contained (if the component doesn't have subcomponents)
-        if (sourceContainer.equals(sourceContained) && ComponentExt.isActor(sourceContained)) {
+      }
+      // if the container is the same as contained (if the component doesn't have subcomponents)
+      if (sourceContainer.equals(sourceContained)
+          && (ComponentExt.isActor(sourceContained) || isSystemOrLogicalSystem(sourceContained))) {
+        if (!isContainedByALogicalComponent(targetContained))
           doCreatePhysicalLink(componentExchange, sourceContainer, (Component) targetContained, port);
-        } else if (ComponentExt.isActor(targetContained) && targetContained.eContainer() instanceof Component) {
-          Component targetContainer = (Component) targetContained.eContainer();
-          doCreatePhysicalLink(componentExchange, sourceContainer, targetContainer);
-        }
+      } else if (ComponentExt.isActor(targetContained) && targetContained.eContainer() instanceof Component) {
+        Component targetContainer = (Component) targetContained.eContainer();
+        if (!isContainedByALogicalComponent(targetContainer))
+          doCreatePhysicalLink(componentExchange, sourceContainer, targetContainer, port);
       }
     }
   }
   
-  private PhysicalComponent findDeployingComponent(Component sourceContainer, ComponentExchange componentExchange,
-      PhysicalComponent targetContained) {
-    for (Part partition : ((PhysicalComponent) targetContained).getRepresentingParts()) {
+  private Component findTheTargetNode(ComponentExchange componentExchange, ComponentPort port) {

```

Comment:
can you explain the rules in a comment ?
thanks

Reply:
sure